cg/apps/cgstats/stats.py

```diff
@@ -142,9 +142,17 @@ def flow_cell_reads_and_q30_summary(self, flow_cell_name: str) -> Dict[str, Unio
         return flow_cell_reads_and_q30_summary
 
     def fastqs(self, flowcell: str, sample_obj: Sample) -> Iterator[Path]:
-        """Fetch FASTQ files for a sample."""
+        """Retrieve all fastq files for a specific sample in a flow cell directory."""
+
+        # The default structure for flow cells demultiplexed with bcl2fastq
         base_pattern = "*{}/Unaligned*/Project_*/Sample_{}/*.fastq.gz"
+
+        # Alternative structure for flow cells demultiplexed with bcl2fastq where the sample fastq files have a trailing sequence
         alt_pattern = "*{}/Unaligned*/Project_*/Sample_{}_*/*.fastq.gz"
-        for fastq_pattern in (base_pattern, alt_pattern):
+
+        # The default structure for flow cells demultiplexed with bclconvert
+        bcl_convert_pattern = "*{}/Unaligned*/*/Sample_{}_*/*.fastq.gz"

```

Comment:
Looking at the example @karlnyr provided I think the pattern will not catch them. Here's an example path: `/home/proj/production/demultiplexed-runs/230822_A00621_0892_AHGTHYDSX7/Unaligned/151250/ACC12542A6_S14_L001_R1_001.fastq.gz`. @karlnyr is this what we want to support?

Reply:
yes - that is what I want to support

Reply:
Adjusted the pattern, should  match the provided example path now

Reply:
This looks good 🔥